### ISSUE TITLE
provider.alicloud: unexpected EOF

### ISSUE THREAD

**Issue by bittopaz** [alibaba/terraform-provider#196]
I have got the error below so many times when run `terraform plan`:

```sh
➜  cn_shanghai_beta git:(master) ✗ tf plan
[WARN] /usr/local/bin/terraform-provider-alicloud overrides an internal plugin for alicloud-provider.
  If you did not expect to see this message you will need to remove the old plugin.
  See https://www.terraform.io/docs/internals/internal-plugins.html
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.template_file.user_data: Refreshing state...
alicloud_security_group.cc: Refreshing state... (ID: sg-uf60v0csy0zj4l4xo62g)
Error refreshing state: 6 error(s) occurred:

* module.sg_web.provider.alicloud: unexpected EOF
* module.vpc.provider.alicloud: unexpected EOF
* module.bastion.provider.alicloud: unexpected EOF
* alicloud_security_group.cc: 1 error(s) occurred:

* alicloud_security_group.cc: alicloud_security_group.cc: unexpected EOF
* module.sg_service.provider.alicloud: unexpected EOF
* module.sg_gitlab_runner.provider.alicloud: unexpected EOF
```

I have to try 3 or 4 times and then get a successful `terraform plan`.

And when in the debug mode I got the errors below, I'm not a Golang developer, not sure if the below error message could help or not. 

```sh
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: goroutine 220 [IO wait]:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net.runtime_pollWait(0x1e81e28, 0x72, 0xc)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/runtime/netpoll.go:164 +0x59
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net.(*pollDesc).wait(0xc420055a38, 0x72, 0x1beb640, 0x1be6570)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x38
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net.(*pollDesc).waitRead(0xc420055a38, 0xc4204eb400, 0x400)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x34
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net.(*netFD).Read(0xc4200559d0, 0xc4204eb400, 0x400, 0x400, 0x0, 0x1beb640, 0x1be6570)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/fd_unix.go:250 +0x1b7
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net.(*conn).Read(0xc4201b24f8, 0xc4204eb400, 0x400, 0x400, 0x0, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/net.go:181 +0x70
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: crypto/tls.(*block).readFromUntil(0xc420604f30, 0x1e8d1d8, 0xc4201b24f8, 0x5, 0xc4201b24f8, 0xc4202648a0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/crypto/tls/conn.go:488 +0x98
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: crypto/tls.(*Conn).readRecord(0xc420500380, 0x182a817, 0xc4205004a0, 0x1051bc0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/crypto/tls/conn.go:590 +0xc4
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: crypto/tls.(*Conn).Read(0xc420500380, 0xc4207de000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/crypto/tls/conn.go:1134 +0x11d
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/http.(*persistConn).Read(0xc4201ce6c0, 0xc4207de000, 0x1000, 0x1000, 0x182a8c8, 0x0, 0xc420264b80)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/http/transport.go:1316 +0x14b
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: bufio.(*Reader).fill(0xc420790f60)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/bufio/bufio.go:97 +0x117
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: bufio.(*Reader).Peek(0xc420790f60, 0x1, 0xc4207910e0, 0xc420024480, 0x20, 0x20, 0x1)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/bufio/bufio.go:129 +0x67
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/http.(*persistConn).readLoop(0xc4201ce6c0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/http/transport.go:1474 +0x196
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: created by net/http.(*Transport).dialConn
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/http/transport.go:1117 +0xa35
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: goroutine 201 [select]:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/yamux.(*Stream).Read(0xc420748340, 0xc420794000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/yamux/stream.go:130 +0x2af
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: bufio.(*Reader).Read(0xc42049ef60, 0xc4205e2f30, 0x1, 0x9, 0xc4205e34b0, 0x194, 0xc4205e3490)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/bufio/bufio.go:213 +0x312
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: io.ReadAtLeast(0x1be7d40, 0xc42049ef60, 0xc4205e2f30, 0x1, 0x9, 0x1, 0x194, 0xc4205e34b0, 0x3)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/io/io.go:307 +0xa9
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: io.ReadFull(0x1be7d40, 0xc42049ef60, 0xc4205e2f30, 0x1, 0x9, 0x1029476, 0x0, 0x4)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/io/io.go:325 +0x58
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: encoding/gob.decodeUintReader(0x1be7d40, 0xc42049ef60, 0xc4205e2f30, 0x9, 0x9, 0xc42076da70, 0x1028dae, 0xc400000008, 0xc4203d3980)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/encoding/gob/decode.go:119 +0x63
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: encoding/gob.(*Decoder).recvMessage(0xc420751e00, 0xc42076da80)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/encoding/gob/decoder.go:76 +0x57
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: encoding/gob.(*Decoder).decodeTypeSequence(0xc420751e00, 0x182a800, 0xc420751e00)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/encoding/gob/decoder.go:140 +0x167
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: encoding/gob.(*Decoder).DecodeValue(0xc420751e00, 0x16c0f80, 0xc42075de80, 0x16, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/encoding/gob/decoder.go:208 +0xe8
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: encoding/gob.(*Decoder).Decode(0xc420751e00, 0x16c0f80, 0xc42075de80, 0xc42000e9f0, 0xc4201ab460)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/encoding/gob/decoder.go:185 +0x16d
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc4205b59e0, 0xc42075de80, 0x17941c0, 0xc4205b5fb0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:407 +0x45
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/rpc.(*Server).readRequestHeader(0xc4201ab440, 0x1bef640, 0xc4205b59e0, 0xc42000e9f0, 0x16, 0xc42075d720, 0x17c2a01, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:580 +0x6c
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/rpc.(*Server).readRequest(0xc4201ab440, 0x1bef640, 0xc4205b59e0, 0xc4201ab440, 0xc4205e2f50, 0xc420751780, 0xc42075d720, 0x179f500, 0xc4205b5fb0, 0x16, ...)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:547 +0x5a
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/rpc.(*Server).ServeCodec(0xc4201ab440, 0x1bef640, 0xc4205b59e0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:466 +0x77
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/rpc.(*Server).ServeConn(0xc4201ab440, 0x2ca0000, 0xc420748340)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:458 +0x373
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/go-plugin.serve(0x2ca0000, 0xc420748340, 0x17fcb99, 0x6, 0x17bc180, 0xc42075d6a0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/go-plugin/rpc_server.go:184 +0x157
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/go-plugin.(*dispenseServer).Dispense.func1(0xc42021a120, 0xc400000012, 0xc4205e2bf8, 0x8, 0x17bc180, 0xc42075d6a0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/go-plugin/rpc_server.go:171 +0x199
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: created by github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/go-plugin.(*dispenseServer).Dispense
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/vendor/github.com/hashicorp/go-plugin/rpc_server.go:172 +0x1d1
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: goroutine 184 [runnable]:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: reflect.(*rtype).Field(0x17cfa60, 0x0, 0x166214f, 0x6, 0x0, 0x0, 0x1bf4a00, 0x16d9a20, 0x0, 0x0, ...)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/reflect/type.go:957 +0x14d
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/util.setQueryValues(0x17cfa60, 0xc4201e7600, 0xc420761458, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/util/encoding.go:64 +0x3fe
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/util.SetQueryValues(0x17cfa60, 0xc4201e7600, 0xc420761458)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/util/encoding.go:39 +0x51
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/util.ConvertToQueryValues(0x17cfa60, 0xc4201e7600, 0x17cfa60)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/util/encoding.go:33 +0x74
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/common.(*Client).Invoke(0xc4202d0360, 0x1805013, 0xf, 0x16b1440, 0x1c3fa88, 0x16b1480, 0xc420597890, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/common/client.go:122 +0x136
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/ecs.(*Client).DescribeRegions(0xc4202d0360, 0x1a, 0x1, 0x1800195, 0x9, 0xc42028d700)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/denverdino/aliyungo/ecs/regions.go:28 +0x10f
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/alicloud.(*Config).ecsConn(0xc420761a00, 0x0, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/alicloud/config.go:134 +0x10b
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/alicloud.(*Config).Client(0xc420761a00, 0x17fd6af, 0x6, 0x16d9a20)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/alicloud/config.go:50 +0x64
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/alicloud.providerConfigure(0xc42015cd20, 0xc42015cd20, 0xc420596cc0, 0xc42062eec0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/alicloud/provider.go:112 +0x1a8
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Configure(0xc4201b0230, 0xc420596cc0, 0xc42062a080, 0xc42003ab10)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:223 +0xed
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Configure(0xc42075d140, 0xc420596cc0, 0xc420248468, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/Users/heguimin/Projects/go/src/github.com/alibaba/terraform-provider/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:478 +0x42
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: reflect.Value.call(0xc420019620, 0xc4202ba110, 0x13, 0x17fb1c2, 0x4, 0xc42003af20, 0x3, 0x3, 0x0, 0x0, ...)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/reflect/value.go:434 +0x91f
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: reflect.Value.Call(0xc420019620, 0xc4202ba110, 0x13, 0xc42077a720, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/rpc.(*service).call(0xc42026a680, 0xc42026a640, 0xc42021aab8, 0xc42026d180, 0xc42062a080, 0x179f500, 0xc420596cc0, 0x16, 0x16b8080, 0xc420248468, ...)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:387 +0x144
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: created by net/rpc.(*Server).ServeCodec
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/rpc/server.go:481 +0x404
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: goroutine 190 [select]:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net/http.(*persistConn).writeLoop(0xc420754480)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/http/transport.go:1704 +0x43a
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: created by net/http.(*Transport).dialConn
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/http/transport.go:1118 +0xa5a
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: goroutine 236 [select]:
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: net.(*netFD).connect.func2(0x1bef380, 0xc420707020, 0xc42011ac40, 0xc4207071a0, 0xc420707140)
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/fd_unix.go:133 +0x1d5
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: created by net.(*netFD).connect
2017/08/07 20:13:07 [DEBUG] plugin: terraform-provider-alicloud: 	/usr/local/go/src/net/fd_unix.go:144 +0x239
2017/08/07 20:13:07 [DEBUG] plugin: /usr/local/bin/terraform-provider-alicloud: plugin process exited
2017/08/07 20:13:07 [DEBUG] plugin: /usr/local/bin/terraform: plugin process exited
```

**Comment by xiaozhu36** [comment]
Hi @bittopaz 

Thank you for your feedback. I can't locate accurately the reason of above issue. Could you please give me your template or parts of it and help me to resolve it as soon as possible, thanks.

Regards,
Guimin

**Comment by bittopaz** [comment]
@xiaozhu36  I use terraform modules a lot. I have been using Terraform for more than half a year with AWS, but I did not meet this kind of errors so often. This happens when refreshing the remote state, I use AWS S3 as my remote backend./

Here is the code I have below:

```sh
module "sg_gitlab_runner" {
  source = "../../../modules/alicloud/alicloud_sg/sg_worker"

  name = "${var.vpc_name}_gitlab_runner"
  vpc_id = "${module.vpc.vpc_id}"
  source_security_groups = ["${alicloud_security_group.cc.id}"]
}

module "sg_service" {
  source = "../../../modules/alicloud/alicloud_sg/sg_service"

  name = "${var.vpc_name}_service"
  vpc_id = "${module.vpc.vpc_id}"
  source_security_groups_to_ssh = ["${alicloud_security_group.cc.id}"]
  source_cidr_blocks_to_ssh = ["1.1.1.1/32"]
  source_cidr_blocks_to_service = ["${var.vpc_cidr}"]
}
```

And the modules:

```sh
// sg_worker
resource "alicloud_security_group" "main" {
  name = "${var.name}"
  vpc_id = "${var.vpc_id}"
}

resource "alicloud_security_group_rule" "ingress_allow_tcp_22_sg" {
  count             = "${length(var.source_security_groups)}"
  type              = "ingress"
  ip_protocol       = "tcp"
  nic_type          = "intranet"
  policy            = "accept"
  port_range        = "22/22"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  source_security_group_id = "${element(var.source_security_groups, count.index)}"
}

resource "alicloud_security_group_rule" "ingress_allow_tcp_22_cidr" {
  count             = "${length(var.source_cidr_blocks)}"
  type              = "ingress"
  ip_protocol       = "tcp"
  nic_type          = "intranet"
  policy            = "accept"
  port_range        = "22/22"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  cidr_ip           = "${element(var.source_cidr_blocks, count.index)}"
}

resource "alicloud_security_group_rule" "egress_allow_all" {
  type              = "egress"
  ip_protocol       = "all"
  policy            = "accept"
  port_range        = "-1/-1"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  cidr_ip           = "0.0.0.0/0"
}

```


sg_service
```sh
resource "alicloud_security_group" "main" {
  name = "${var.name}"
  vpc_id = "${var.vpc_id}"
}

resource "alicloud_security_group_rule" "ingress_allow_sg_tcp_22" {
  count             = "${length(var.source_security_groups_to_ssh)}"
  type              = "ingress"
  ip_protocol       = "tcp"
  nic_type          = "intranet"
  policy            = "accept"
  port_range        = "22/22"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  source_security_group_id = "${element(var.source_security_groups_to_ssh, count.index)}"
}

resource "alicloud_security_group_rule" "ingress_allow_cidr_tcp_22" {
  count             = "${length(var.source_cidr_blocks_to_ssh)}"
  type              = "ingress"
  ip_protocol       = "tcp"
  nic_type          = "intranet"
  policy            = "accept"
  port_range        = "22/22"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  cidr_ip           = "${element(var.source_cidr_blocks_to_ssh, count.index)}"
}

resource "alicloud_security_group_rule" "ingress_allow_service" {
  count             = "${length(var.source_cidr_blocks_to_service)}"
  type              = "ingress"
  ip_protocol       = "tcp"
  nic_type          = "intranet"
  policy            = "accept"
  port_range        = "40000/40080"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  cidr_ip           = "${element(var.source_cidr_blocks_to_service, count.index)}"
}

# Ingress allow tcp 8000
resource "alicloud_security_group_rule" "ingress_allow_cidr_tcp_80000" {
  count             = "${length(var.source_cidr_blocks_to_80000)}"
  type              = "ingress"
  ip_protocol       = "tcp"
  nic_type          = "intranet"
  policy            = "accept"
  port_range        = "80000/80000"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  cidr_ip           = "${element(var.source_cidr_blocks_to_8000, count.index)}"
}

resource "alicloud_security_group_rule" "egress_allow_all" {
  type              = "egress"
  ip_protocol       = "all"
  policy            = "accept"
  port_range        = "-1/-1"
  priority          = 1
  security_group_id = "${alicloud_security_group.main.id}"
  cidr_ip           = "0.0.0.0/0"
}

```

If you need more info to debug, please let me know, I'd love to help.

**Comment by xiaozhu36** [comment]
Hi @bittopaz 

I'm sorry for you have happened above issue, and thank you for your info. 

I have made a test according to your template and there is good. Can you tell me which version of terraform-provider-alicloud you used? Thanks.

By the way, your template has the following issues:
1. Missing module vpc and resource alicloud_security_group.cc, and you should add them before running;
2. The port_range of resource "alicloud_security_group_rule.ingress_allow_cidr_tcp_80000" is invalid, the valid value of port_range is 1~65535. Please check it;
3.  The cidr_ip of resource "alicloud_security_group_rule.ingress_allow_cidr_tcp_80000"  should use variable source_cidr_blocks_to_80000 and not source_cidr_blocks_to_8000. Please check it.

Looking forward to your feedback, thanks.

Regards,
Guimin

**Comment by bittopaz** [comment]
@xiaozhu36 thanks for pointing this.

This is only part of my terraform files. I could run them successfully. 
For the port part, sorry for the confusing, I changed the port number after paste the content here. 

I'm using version v1.1.6

**Comment by bittopaz** [comment]
@xiaozhu36 Are you able to reproduce this?

This is really annoying, I get 3 failures and then got a success `terraform plan`. And then 3 times failure for `terraform apply`...

**Comment by xiaozhu36** [comment]
@bittopaz 

I'm sorry for that. There is no progress for it according to above error, and I can't reappear your error. My test is ok using version v1.1.6. The latest error is still same as above?

**Comment by bittopaz** [comment]
@xiaozhu36 I get some inspiration form https://github.com/hashicorp/terraform/issues/8011. 

After set `-parallelism=1`, it is stable now...

Hope this info could help.

**Comment by xiaozhu36** [comment]
@bittopaz Thank you for your inspiration, and I will check and resolve it by referring to it. Thanks.

**Comment by bittopaz** [comment]
@xiaozhu36  Do we have an update on this issue, please?

**Comment by bittopaz** [comment]
I can confirm, this is fixed when I use terraform 0.11.1 and alicloud-provider 1.2.11
